robots/pkg/test-label-analyzer/stats.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package test_label_analyzer
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// FileStats contains the information of the file whose outline was traversed and the results of the
+// traversal.
+type FileStats struct {
+	*Config    `json:"config"`
+	*TestStats `json:"test_stats"`
+
+	// RemoteURL is the absolute path to the file, most certainly an absolute URL inside a version control repository
+	// containing a commit ID in order to exactly define the state of the file that was traversed
+	RemoteURL string `json:"path"`
+}
+
+// TestStats contains the results of traversing a set of Ginkgo outlines and collecting the pathes in the form of
+// GinkgoNode slices matching a Config describing the criteria to match against.
+type TestStats struct {
+
+	// SpecsTotal is the total number of specs encountered during traversal of the outline
+	SpecsTotal int `json:"specs_total"`
+
+	// MatchingSpecPathes is the slice of PathStats to matching specs for the collection of outlines traversed.
+	// Each PathStats inside this slice is the path to each of the matching specs defined by the Config
+	// being used. Please note that the GinkgoNode.Nodes are being removed during traversal.
+	MatchingSpecPathes []*PathStats `json:"matching_spec_pathes"`
```

Comment:
nit: 
```suggestion
	MatchingSpecPaths []*PathStats `json:"matching_spec_paths"`
```
In case of the acceptance, keep in mind that we need to edit the README as well